mesonbuild/backend/backends.py

```diff
@@ -455,9 +455,12 @@ def get_external_rpath_dirs(self, target):
                 args.extend(self.environment.coredata.get_external_link_args(target.for_machine, lang))
             except Exception:
                 pass
+        # Match -Wl,-rpath= and -Wl,-rpath, formats
+        rpath_regex = re.compile(r'-Wl,-rpath?[=,]([^,@\-][^,]+)')
```

Comment:
The regex looks overly complicated still, and I'm not sure the h should be optional.

Can't it just be something like r'-Wl-rpath[=,](.*)' ?  If not, *that* is what the comment should explain.

Reply:
> I'm not sure the h should be optional

Yeah, had forgotten to remove that, updated.

> Can't it just be something like r'-Wl-rpath=,' ? If not, that is what the comment should explain.

Well the rest is just the capture group for matching the actual rpath paths that come after `-Wl,-rpath=` or `-Wl,-rpath,` args.

Reply:
My comment may not have come through properly.  Why is the capture group pattern so complicated?  Shouldn't it just be (.*) ?

Reply:
So parsing this correctly seems to be significantly more complex than I was expecting, it appears [this](https://github.com/bminor/binutils-gdb/blob/binutils-2_34/ld/lexsup.c#L1070-L1148) is the way ld parses the options which I suppose we want to try and fully emulate.

This one seems particularly annoying since we would have to actually do a path lookup:
```
For compatibility with other ELF linkers, if the -R option is followed by a directory name, rather than a file name, it is treated as the -rpath option.
```

Reply:
Oh, man, I forgot about that case.

Frankly, I would just fix the cases buildroot tickles, and file a followon bug for the others...

Reply:
Well I've got that one added with a path check and it seems to pass in tests.